src/huak/env/venv.rs

```diff
@@ -249,6 +249,37 @@ impl Venv {
         Ok(())
     }
 
+    /// Run a command in the context of the venv.
+    pub fn exec_command(&self, command: &str) -> HuakResult<()> {
+        // Create the venv if it doesn't exist.
+        // TODO: Fix this.
+        self.create()?;
+
+        let source_command = get_shell_source_command()?;
+        let script = self.get_activation_script()?;
+        let activation_command =
+            format!("{} {}", source_command, script.display());
+
+        let shell_path = get_shell_path()?;
+        let mut new_shell = expectrl::spawn(&shell_path)?;
+        new_shell.send_line(&activation_command)?;
+        new_shell.send_line(command)?;
+        new_shell.send_line("exit")?;
+
+        if OS == "windows" {
+            let stdin = expectrl::stream::stdin::Stdin::open()?;
+            new_shell.interact(stdin, std::io::stdout()).spawn()?;
+            let stdin = expectrl::stream::stdin::Stdin::open()?;
+            stdin.close()?;
+        } else {
+            let mut stdin = expectrl::stream::stdin::Stdin::open()?;
+            new_shell.interact(&mut stdin, std::io::stdout()).spawn()?;
+            stdin.close()?;
+        }
```

Comment:
You should be able to do something like this instead:
```rust
        let shell_path = get_shell_path()?;
        let cmd = std::process::Command::new(shell_path)
            .args(["-c", &format!("{} && {}", activation_command, command)])
            .output()
            .unwrap();
        print!("{}", String::from_utf8(cmd.stdout).unwrap());
        // and same for stderr
        // and probably exit with exit code of the command

```
Only assumption here is that shell takes `-c` argument which takes the command to be ran. IIRC most shells have that, popular weird ones to verify that would be powershell and fish. And the `&&` operand, but powershell does support that, so I expect every shell to support that.

There might be a better way. You could check the source code of poetry to see how they do it (and borrow (not mutably! ;]) some ideas)

Reply:
Nice! I didn't know we already had a function for that